corehq/apps/locations/fixtures.py

```diff
@@ -207,14 +211,47 @@ def should_sync_flat_fixture(project):
 )
 
 
+int_field = IntegerField()
+int_array = ArrayField(int_field)
+
+
 def get_location_fixture_queryset(user):
     if toggles.SYNC_ALL_LOCATIONS.enabled(user.domain):
         return SQLLocation.active_objects.filter(domain=user.domain).prefetch_related('location_type')
 
     timing = TimingContext("get_location_fixture_queryset")
     with timing("mptt"):
         mptt_set = mptt_get_location_fixture_queryset(user)
-    return ComparedQuerySet(mptt_set, timing)
+    with timing("cte"):
+        cte_set = cte_get_location_fixture_queryset(user)
+    return ComparedQuerySet(mptt_set, cte_set, timing)
+
+
+def cte_get_location_fixture_queryset(user):
+
+    user_locations = user.get_sql_locations(user.domain)
+
+    if user_locations.query.is_empty():
+        return user_locations
+
+    fixture_ids = With(raw_cte_sql(
+        """
+        SELECT "id", "path", "depth"
+        FROM get_location_fixture_ids(%s::TEXT, %s)
```

Comment:
@esoergel I moved the main logic of this function into a SQL function. It is a big function, but I think the SQL is easier to read (and more efficient) than the ORM code was, and should dramatically cut down on the size of SQL sent from django to postgres.

Aside: had to make a new construct ([`raw_cte_sql`](https://github.com/dimagi/django-cte/blob/master/django_cte/raw.py#L5)) to enable joining to a table function. I didn't see a way to do that with stock django.